src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,401 @@
+        String minCorrectPassword = "aA1";
+        correctPasswords.add(minCorrectPassword);
+
+        getSpecialCharacters("\\").forEach(
+                specialChar -> correctPasswords.add(minCorrectPassword.replaceAll(
+                        REGROUPED_PASSWORD_PATTERN, specialChar.concat("$1")))
+        );
+
+        return correctPasswords;
+    }
+
+    private static List<String> createWrongPasswords() {
+        String minCorrectPassword = "aA1";
+        List<String> wrongPasswords = new ArrayList<>(List.of(
+                "",
+                " ",
+                "w",
+                "1",
+                "W",
+                "`"
+        ));
+
+        List<String> tempList = new ArrayList<>();
+        IntStream.range(0, wrongPasswords.size()).forEach(i -> {
+            for (String wrongPassword : wrongPasswords) {
+                tempList.add(wrongPasswords.get(i).concat(wrongPassword));
+                tempList.add(wrongPasswords.get(i).concat(wrongPassword).concat(wrongPasswords.get(i)));
+            }
+        });
+
+        wrongPasswords.addAll(tempList);
+
+        wrongPasswords.addAll(List.of(
+                minCorrectPassword.concat(" "),
+                " ".concat(minCorrectPassword),
+                minCorrectPassword.concat(" A"),
+                "A ".concat(minCorrectPassword)
+        ));
+
+        getRussianAndUkrLetters().forEach(letter ->
+                createCorrectPasswords().forEach(password -> wrongPasswords.add(
+                        password.replaceAll(REGROUPED_PASSWORD_PATTERN, letter.concat("S1")))));
+
+        return wrongPasswords;
+    }
+
+    private static List<String> getCorrectPhoneNumbers() {
+        List<String> correctPhoneNumbers = new ArrayList<>();
+
+        String correctNumber = "+12-345-678-90-12";
+        String correctNumber1 = "13.345.678.90.13";
+
+        correctPhoneNumbers.add(correctNumber);
+        correctPhoneNumbers.add(correctNumber1);
+
+        return correctPhoneNumbers;
+    }
+
+    private static List<String> createCorrectPhoneNumbers() {
+        List<String> correctPhoneNumbers = getCorrectPhoneNumbers();
+        List<String> replacementList = List.of(
+                "   $1-$2-$3-$4-$5-$6-$7-$8-$9   ",
+                "   $1.$2.$3.$4.$5.$6.$7.$8.$9   ",
+                "$1$3$5$7$9",
+                "$1     $3     $5     $7     $9",
+                "$1$2($3)$4$5$6$7$8$9",
+                "$1($3)$5$7$9",
+                "$1(.- $3 -.)$5 -.- $7 -.- $9",
+                "$1$2(   $3   )$4$5$6$7$8$9",
+                "$1$2((($3)))$4$5$6$7$8$9",
+                "$1$2(((((($3.-.-.)$4$5$6$7$8$9",
+                "$1$2(---$3...)$4$5$6$7$8$9");
+
+        List<String> tempList = new ArrayList<>();
+        for (String correctNumber : correctPhoneNumbers) {
+            addReplacedPhoneNumbers(replacementList, tempList, correctNumber);
+        }
+        correctPhoneNumbers.addAll(tempList);
```

Comment:
Lines 199 - 203 needs to be refactored. Such code have loop and loop inside the method... Why don't you make only one loop?